src/redux/books/books.js

```diff
@@ -0,0 +1,42 @@
+const ADD = 'bookstore/books/ADD';
+const REMOVE = 'bookstore/books/REMOVE';
+const UPDATE = 'bookstore/books/UPDATE';
+
+const addBook = (book) => ({
+  type: ADD,
+  book,
+});
+
+const removeBook = (book) => ({
+  type: REMOVE,
+  book,
+});
+
+const updateBook = (book) => ({
+  type: UPDATE,
+  book,
+});
+
+const bookReducer = (state = [], action = {}) => {
+  switch (action.type) {
+    case ADD:
+      return [
+        ...state,
+        action.book,
+      ];
+    case REMOVE:
+      return state.filter((book) => book.id !== action.book.id);
+    case UPDATE:
+      return state.map((book) => {
+        if (book.id === action.book.id) {
+          return action.book;
+        }
+        return book;
+      });
+    default:
+      return state;
+  }
+};
+
+export { addBook, updateBook, removeBook };
+export default bookReducer;
```

Comment:
- [ ] Good job so far to implement some of the requirements requested :+1: But from the project requirement, you need to first set an initial state with an empty array. then creates and Exports Action Creators. (addBook and removeBook functions) and reducers as **default**  and you need to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux). Please fix it :smile: . See the perfect example below :arrow_down: 
```
// widgets.js

// Actions
const LOAD   = 'my-app/widgets/LOAD';
const CREATE = 'my-app/widgets/CREATE';
const UPDATE = 'my-app/widgets/UPDATE';
const REMOVE = 'my-app/widgets/REMOVE';

// Reducer
export default function reducer(state = {}, action = {}) {
  switch (action.type) {
    // do reducer stuff
    default: return state;
  }
}

// Action Creators
export function loadWidgets() {
  return { type: LOAD };
}

export function createWidget(widget) {
  return { type: CREATE, widget };
}

export function updateWidget(widget) {
  return { type: UPDATE, widget };
}

export function removeWidget(widget) {
  return { type: REMOVE, widget };
}

// side effects, only as applicable
// e.g. thunks, epics, etc
export function getWidget () {
  return dispatch => get('/widget').then(widget => dispatch(updateWidget(widget)))
}
```